common/src/main/java/com/wynntils/wynn/model/CharacterManager.java

```diff
@@ -80,14 +86,27 @@ public static void onWorldStateChanged(WorldStateEvent e) {
             inCharacterSelection = true;
         }
 
-        if (e.getNewState() == WorldStateManager.State.WORLD
-                && e.getOldState() != WorldStateManager.State.CHARACTER_SELECTION) {
-            // We went directly to a world without coming from the character selection
-            // menu. This means the player has "autojoin" enabled, and that we did not
-            // get a chance to read the character info from the character selection menu.
-            // Instead, we send a container query to read it from the character (compass) menu.
-            WynntilsMod.info("Scheduling character info query");
-            scanCharacterInfoPage();
+        if (e.getNewState() == WorldStateManager.State.WORLD) {
+            if (e.getOldState() != WorldStateManager.State.CHARACTER_SELECTION) {
+                // We went directly to a world without coming from the character selection
+                // menu. This means the player has "autojoin" enabled, and that we did not
+                // get a chance to read the character info from the character selection menu.
+                // Instead, we send a container query to read it from the character (compass) menu.
+                WynntilsMod.info("Scheduling character info query");
+
+                // This time, we need to scan character info and profession info as well.
+                scanCharacterInfoPage();
+            } else {
+                // We did not auto-join, we have a correct ID already.
+                int oldId = currentCharacter.id;
+                scanCharacterInfoPage();
```

Comment:
This means we will always do `scanCharacterInfoPage`, which is calling the container query API ("fake inventories"). I understood there were some hesitations about the stability of this, but maybe those fears have been alleviated by now..?

Reply:
We need to do this so profession info is calculated. The query itself seemes to be pretty stable.

Reply:
Yes, I understand why. I was just pointing out that when I started doing this (for auto-join) I met with opposition for doing it always. (I think we should do it always, like you do now.)